packages/salesforcedx-vscode-apex/src/commands/apexTestRunCodeAction.ts

```diff
@@ -153,6 +154,60 @@ export class ApexLibraryTestRunExecutor extends LibraryCommandletExecutor<{}> {
     );
     return new vscode.Range(pos, pos);
   }
+
+  private async mapApexArtifactToFilesystem(
+    tests: ApexTestResultData[],
+    packageDirectories: NamedPackageDir[]
+  ): Promise<Map<string, string>> {
+    // Create a map to store the correlated artifacts
+    const correlatedArtifacts: Map<string, string> = new Map();
+
+    // Iterate over each test in the ApexTestResultData array
+    for (const test of tests) {
+      // Get the name of the test without the extension
+      const testName = test.apexClass.fullName ?? test.apexClass.name;
+      // Add the test name to the set
+      correlatedArtifacts.set(testName, 'unknown');
+    }
+
+    const workspaceFolder = vscode.workspace.workspaceFolders
+      ? vscode.workspace.workspaceFolders[0]
+      : undefined;
+    if (!workspaceFolder) {
+      return correlatedArtifacts;
+    }
+
+    const patterns = packageDirectories.map(pkgDir => {
+      const relativePath = path.relative(
+        workspaceFolder.uri.fsPath,
+        pkgDir.fullPath
+      );
+      return `${relativePath}/**/*.cls`;
+    });
+
+    const findFilesPromises = patterns.map(pattern => {
+      return vscode.workspace.findFiles(pattern, '**/node_modules/**');
+    });
+
+    const filesWithDuplicates = (await Promise.all(findFilesPromises)).flat();
```

Comment:
These two seem like good candidates for going in a shared utilty

Reply:
I'll have a look